getGeocode.js

```diff
@@ -0,0 +1,66 @@
+const dotenv = require("dotenv");
+//note: this only works locally; in Lambda we use environment variables set manually
+dotenv.config();
+
+const fetch = require("node-fetch");
+const { Client } = require("@googlemaps/google-maps-services-js");
+const { generateKey } = require("./data/dataDefaulter");
+
+const getGeocode = async (name, street, zip) => {
+    const address = `${name},MA,${street},${zip}`;
+    const client = new Client();
+    try {
+        const resp = await client.geocode({
+            params: {
+                address,
+                key: process.env.GOOGLE_API_KEY,
+            },
+        });
+        return resp.data;
+    } catch (e) {
+        console.error(e.response.data);
+    }
+};
+
+const sleep = (milliseconds) => {
+    return new Promise((resolve) => setTimeout(resolve, milliseconds));
+};
+
+const getAllCoordinates = async (locations, cachedResults) => {
+    const existingLocations = cachedResults.reduce((acc, location) => {
+        const { resolvedLocation, latitude, longitude } = location;
+        if (resolvedLocation && latitude && longitude) {
+            acc[generateKey(location)] = {
+                resolvedLocation,
+                latitude,
+                longitude,
+            };
+            return acc;
+        } else {
+            return acc;
+        }
+    }, {});
+
+    const coordinateData = await Promise.all(
+        locations.map(async (location) => {
+            const { name = "", street = "", zip = "" } = location;
+            const locationInd = generateKey(location);
+
+            if (existingLocations[locationInd]) {
+                return { ...location, ...existingLocations[locationInd] };
+            } else {
+                const locationData = await getGeocode(name, street, zip);
+                await sleep(100);
+                return {
+                    ...location,
+                    resolvedLocation: locationData.results[0].formatted_address,
```

Comment:
Do we need this data for anything? Will we use it on the front-end?

Reply:
I mainly kept this in there in case we need to check if a location is correctly resolving. Easier to check a bunch of formatted addresses than a bunch of lat/long coordinates in the S3 data.

Reply:
Gotcha. I'm that case could you please remove it for now? We have the address on other fields in this object so I think we'll be good with that, and we can always add it back in if we need to debug locally.